src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5952,14 +5953,15 @@ private static string GetNamespaceToRemove(CompletionContext context, TypeComple
         internal static List<CompletionResult> CompleteHelpTopics(CompletionContext context)
         {
             var results = new List<CompletionResult>();
-            var dirPath = Utils.GetApplicationBase(Utils.DefaultPowerShellShellID) + "\\" + CultureInfo.CurrentCulture.Name;
+            var dirPath = Utils.GetApplicationBase(Utils.DefaultPowerShellShellID) + Path.DirectorySeparatorChar + CultureInfo.CurrentCulture.Name;
             var wordToComplete = context.WordToComplete + "*";
             var topicPattern = WildcardPattern.Get("about_*.help.txt", WildcardOptions.IgnoreCase);
             string[] files = null;
 
             try
             {
-                files = Directory.GetFiles(dirPath, wordToComplete);
+                var wildcardPattern = WildcardPattern.Get(wordToComplete, WildcardOptions.IgnoreCase);
+                files = Directory.GetFiles(dirPath).Where(f => wildcardPattern.IsMatch(Path.GetFileName(f))).ToArray();
```

Comment:
It's recommended to avoid LINQ in perf-sensitive code path (see [pref-considerations](https://github.com/PowerShell/PowerShell/blob/master/docs/dev-process/coding-guidelines.md#performance-considerations)). Tab completion is perf sensitive, so could you please replace this with a loop?